base/abstractarray.jl

```diff
@@ -2940,6 +2940,11 @@ mapany(f, itr) = Any[f(x) for x in itr]
 Transform collection `c` by applying `f` to each element. For multiple collection arguments,
 apply `f` elementwise, and stop when when any of them is exhausted.
 
+Note that `f` makes no guarantees about the order in which `f` is called on elements. In
+addition, `f` is assumed to be [pure](https://en.wikipedia.org/wiki/Pure_function). Using
+an impure function together with `f` can cause bugs when working with some data structures,
+e.g. sparse or diagonal arrays, as `f` will only be called once on duplicated elements.
```

Comment:
it will be called once on default `0` but if you have duplicates that are non-default zeros it will be called multiple times.

Reply:
> it will be called once on default `0` but if you have duplicates that are non-default zeros it will be called multiple times.

I've edited to say that it "May" call it only once per duplicate element, and clarify that calling `map` with an impure function is undefined behavior.